fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * RealStorageServer.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/MockPeekCursor.h"
+#include "fdbserver/MockLogSystem.h"
+#include "fdbclient/StorageServerInterface.h"
+#include "fdbserver/IKeyValueStore.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "fdbserver/RecoveryState.h"
+#include "fdbserver/ptxn/test/Driver.h"
+
+#include "flow/UnitTest.h"
+
+#include "flow/actorcompiler.h" // has to be last include
+
+namespace ptxn {
+
+ACTOR Future<Void> driverRunStorageServer(StorageServerInterface* ssi,
+                                          std::vector<Future<Void>>* actors,
+                                          Tag tag,
+                                          Reference<MockPeekCursor> mockPeekCursor) {
+
+	KeyValueStoreType storeType = KeyValueStoreType::SSD_BTREE_V2;
+	state std::string folder = ".";
+	std::string fileName = joinPath(folder, "storage-" + ssi->id().toString() + "." + storeType.toString());
+	std::cout << "new Storage Server file name: " << fileName << std::endl;
+	deleteFile(fileName);
+	state IKeyValueStore* data = openKVStore(storeType, fileName, ssi->id(), 0);
+
+	state ReplyPromise<InitializeStorageReply> storageReady;
+
+	// Initialize dbInfo with mocked cursor and mocked loy system.
+
+	// TODO: Use Reference API?
+	// TODO: delete mockLogSystem eventually.
+	state MockLogSystem* mockLogSystem = new MockLogSystem();
+	mockLogSystem->cursor = mockPeekCursor.castTo<ILogSystem::IPeekCursor>();
+
+	LogSystemConfig logSystemConfig;
+	logSystemConfig.logSystemType = LogSystemType::mock;
+	logSystemConfig.mockLogSystem = mockLogSystem;
+
+	ServerDBInfo dbInfoBuilder;
+	dbInfoBuilder.recoveryState = RecoveryState::ACCEPTING_COMMITS;
+	dbInfoBuilder.logSystemConfig = logSystemConfig;
+	state Reference<AsyncVar<ServerDBInfo>> dbInfo = makeReference<AsyncVar<ServerDBInfo>>(dbInfoBuilder);
+	// Initialize dbInfo done.
+
+	std::cout << "Starting Storage Server" << std::endl;
+	state Future<Void> ss = storageServer(data, *ssi, tag, storageReady, dbInfo, folder);
+	actors->emplace_back(ss);
+
+	// Delay 0.1 seconds to wait storage server processing the mutations.
+	// TODO: Wait until certain desired version is reached rather than predefined delay time.
+	wait(delay(0.1));
```

Comment:
I think you can just `wait(ss)` here. At the caller side, you can have something like:

```
loop choose {
	when(wait(driver.runStorageServer())) {
        // print some error
    }
    when(wait(driver.verifyGetValue("Hello"_sr, 8, "World"_sr)))) {
         break;
    }
    when (wait(actors.getResult())) {}
}
```

The `verifyGetValue` may need some adjustment. I.e., if the RPC didn't get the data, because SS hasn't peeked the data yet, then retry a few times. Or, when the SS gets the data, use a Promise to tell `verifyGetValue` that it can send the PRC. See https://github.com/apple/foundationdb/pull/4761/ for examples of such synchronization.